### ISSUE TITLE
Submitting configs for mods.

### ISSUE THREAD

**Issue by Jurkkis** [Mogli12/GearboxAddon#95]
Can users submit configs for mod tractors to be included with the mod?
I've created one which I've IRL experience with.
http://www.farming2015mods.com/farming-simulator-2017-mods/fs-17-tractors/fs17-zetor-9540-10540/

Config:
https://gist.github.com/Jurkkis/fef8daede40a5af5408b5d800265fe9f

**Comment by Quintto** [comment]
i think that would be an amazing idea!

**Comment by Quintto** [comment]
btw I'm working on a gearbox for the steyr 8110, I'm almost done but in which unit is the fuelUsageRatio , i can't find it in the documentation, i assume its in g/kWh, but i'm not sure

**Comment by Mogli12** [comment]
Hi, the unit for fuel usage is g/kWh.

**Comment by Quintto** [comment]
sorry for hijacking this issue thread, but i've added the gearbox stuff to the tractor, but now its not going in game its giving me this error: Register vehicle type: STEYR_8110aTurbo_SK2_V1.STEYR8110SK2
Error: C:/Users/Asus/Documents/My Games/FarmingSimulator2017/mods/STEYR_8110aTurbo_SK2_V1/Steyr8110Turbo.xml. . .
Error: No storeData found in 'C:/Users/Asus/Documents/My Games/FarmingSimulator2017/mods/STEYR_8110aTurbo_SK2_V1/Steyr8110Turbo.xml'. StoreItem will be ignored!

**Comment by Mogli12** [comment]
Hi,

It would be much easier to create a file gearboxAddonConfig.xml in the mods folder. You can put all your configurations there. This works in SP and MP. Only the server needs the gearboxAddonConfig.xml.

Additionally, I propose to use notepad++ to edit XML. By the way: just delete line 32 in Steyr8110Turbo.xml.

Regards, Stefan

**Comment by Quintto** [comment]
thanks for the help, but i always thought you couldn't upload an XML file to a dedicated server right? (im using notepad++ already)

**Comment by Mogli12** [comment]
I do not have access to a dedi server right now.

**Comment by Quintto** [comment]
I do, I will try it tonight

**Comment by Jurkkis** [comment]
Thanks for response. If clients have their own configfile in mod folder, the server one will override?
The configfile sure is easiest way, or include the gearbox config with the mods xml if you edit/maker your own vehicles. Just thought that making the configs takes some time to find the torque/gear ratio info for all your favorite mods, it would be nice to have some database for those.

As for uploading the gearboxAddonConfig.xml to dedicated server mod folder, it wont work through the FS web control panel, it just gives error for not supported mod type. You will need FTP Access to dedi mod folder, and not all server hosters provide this.

**Comment by Quintto** [comment]
I thought so, a database would be a nice idea, maybe make a google drive thats open for everyone to put their config (per tractor) in, like a zip file with: the config, sources and mod link
like this one? https://drive.google.com/drive/folders/0B9A9perqHdPcSTVDSEdBRU9lcTg?usp=sharing

**Comment by Mogli12** [comment]
Nice idea. I have some configurations, which I did not want to include directly in FS17_GearboxAddon. I loaded the XML into your google drive.

**Comment by Quintto** [comment]
but why didn't you want to include them, I can't imagine its because of the size. those extra lines barely add any Kb's

**Comment by Anonymous-any** [comment]
Hi there maybe to modify and share them with easy way for all? because It's a constraint to update inside the zip file cause when online you know all player and server must have the same hash archive file...Although there is a way to bypass certain files from the archives outside but the solution is farfetched

**Comment by Mogli12** [comment]
Please do not modify the mod. Just place the gearboxAddonConfig.xml in the mods folder.

**Comment by Anonymous-any** [comment]
Sure ! that why I suggest to try to put outside configfiles ! if the idea can make his path.

**Comment by Quintto** [comment]
maybe its possible to make a seperate mod that can contain the configfiles, from which the gearboxmod grabs its additional configs

**Comment by cabmac** [comment]
with the gearboxAddonConfig.xml in your MODS-FOLDER you have a seperated file for all your personal vehicles without to touch de gearbox-mod (zip)

**Comment by Quintto** [comment]
yeah but the point is making it work on a dedicated server, where you can't have XML files in the mods folder, only ZIP files

**Comment by cabmac** [comment]
ah, ok,
a database would be very interesting, but i think i has to be in a forum, something like this:
https://vertexdezign.net/forum/index.php/Thread/1616-Austausch-von-Getriebe-f%C3%BCr-Gearbox-Addon-ls15-ls17/?postID=22338&highlight=gearbox#post22338

**Comment by Quintto** [comment]
yeah probably, but maybe make it on an English forum like FS-UK or http://modcentral.co.uk/ so everyone can participate and understand what people are saying

**Comment by Jurkkis** [comment]
Forum sounds good. Someone just makes initial post with template how to post and (upload) configs that everyone follows. I've got 40km/h delta powershift gearing setup ready for older Valtras that is good for Mega/Hitech 8000 series and early T-Series. Just need to make torque curves for different engines.

**Comment by Mogli12** [comment]
Maybe we could combine this one with #89. We need some more flexibility to store configurations, even on the dedi server. Selecting the right transmission out of different possibilities could be part of the configuration menu.
This is just an idea for the next version. Maybe we can even access a central server like in https://www.modhoster.de/mods/global-market? I do not know yet.

**Comment by Quintto** [comment]
if it would be possible to make such central database for all the configs, that would be really nice especially if it could be combined with selectable transmission. 
but for now i think a forum thread would be nice, should I make one on fs-uk?

**Comment by Mogli12** [comment]
@Quintto: Yes!

**Comment by Mogli12** [comment]
Hmm, maybe we can store the config here in github???

**Comment by cabmac** [comment]
could be interesting too,

**Comment by Quintto** [comment]
made the topic: http://fs-uk.com/forum/index.php?topic=191932.new#new

**Comment by cabmac** [comment]
very awesome detailed example...this is how it should be...

**Comment by Quintto** [comment]
thanks, now lets hope people actually help and share their configs

**Comment by Anonymous-any** [comment]
Hi, Quintto is nothing else is losing in you shared config? like the pathmod and xml name to found down in configFile? but maybe i'm wrong in all cases i've dl the tractor and try to do it.
Regards

**Comment by Quintto** [comment]
that's because that config was made to put in the mod.xml and not in a config, so I'm able to put it on my dedicated server

**Comment by Anonymous-any** [comment]
Oh ok sorry but newbies go have a shared config nonfunctional no?

**Comment by Quintto** [comment]
what do you mean?

**Comment by Anonymous-any** [comment]
well to install a config a guy who don't know all base about the mod go have a config not functionnal if he read the post and try (my english is sad no? ^^)

**Comment by Quintto** [comment]
I see what you mean, but I don't really know how I should go about explaining that, maybe @Mogli12 has an idea?

**Comment by Anonymous-any** [comment]
yes i don't know but at time what is see is just adding
		
		<vehicle>STEYR8090aTurboSK2
		<mod name="FS17_STEYR8090aTurboSK2">
			<xmlFile name="Steyr8090TurboSK2.xml"/>	
		</mod>
go solve the problem than it do? add this copy paste  to your share for a beginner seem to work?

**Comment by Anonymous-any** [comment]
Well :/ i try to buy the tractor and returning an error is that your mod or conversion Quintto?

My Games/FarmingSimulator2017/mods/FS17_STEYR8090aTurboSK2/Steyr8090Turbo.i3d (290.56) ms
Error: Running LUA method 'update'.
[builtin:deg]:0: attempt to perform arithmetic on a nil value

**Comment by Quintto** [comment]
thats because its for the steyr 8110 not the 8090

**Comment by Anonymous-any** [comment]
no no I give the explain sorry

**Comment by Anonymous-any** [comment]
No way possible then?

**Comment by Mogli12** [comment]
Hi, you can create two <vehicle> entries for the same mod for different engines. Example is at the end of 
https://github.com/Mogli12/GearboxAddon/blob/master/FS17_GearboxAddon/gearboxAddonConfig.xml.

It works like this:
```xml
	<vehicle>Steyr profi CVT
		<mod name="FS17_SteyrProfi">
			<xmlFile name="SteyrProfiCVT.xml" engines="2 4 6 8"/>
		</mod>
```

**Comment by Quintto** [comment]
nice! 
but I assume you still need to add the extra engine configs in the mod.xml right?
so instead of not touching the xml, you need to add the extra engines for the Profi CVT additionally to the powershift ones?
anyways, this is an awesome addition!

**Comment by Mogli12** [comment]
Yes, you need two engine configurations. So there is one for the power shift transmission and one for the CVT. It is just that the `engines` attribute limits the configuration entry to the corresponding motor configurations.

**Comment by Anonymous-any** [comment]
Cool ! thank and then we can choose to insert the turbo sound

`<blowOffVentilSound volume=`

or not many thank Maestro ! :)

**Comment by Anonymous-any** [comment]
Hi, I would submit a question not sure is there not a bug in the "newHolland8340" range configuration where there is two almost the same like this?
`			<ranges defaultRange="2" shiftTimeMs="800">
				<range name="L" ratio="0.43"/>
				<range name="H" ratio="1"/>
			</ranges>
			
			<ranges defaultRange="2" shiftTimeMs="800">
				<range name="L" ratio="0.245"/>
				<range name="H" ratio="1"/>
			</ranges>`
?? to know if I can remove one that do nothing? but I know nothing ^^
Br

**Comment by Mogli12** [comment]
Hi,
This article is in German. But the Fords and New Hollands of that time have a transmission with 16/16 gears, 4 power shift gears and 2 groups with 2 ranges:
http://www.dlg-test.de/pbdocs/traktoren/ford-8240sle.pdf

**Comment by Anonymous-any** [comment]
Hi, thank it's ok but the two are used in same time or we must choose the one we want to keep?
Br

**Comment by Mogli12** [comment]
Hi,
The keys for the first range are KEY_rightbracket and KEY_backslash. If you additionally press the right shift key you can change the second range.
Regards, Stefan

**Comment by Anonymous-any** [comment]
Why I'm so stupid sorry for disturb big thank you Stephan
